momepy/coins.py

```diff
@@ -0,0 +1,539 @@
+        self.already_merged = True
+        print('>'*50 + ' [%d/%d] '%(len(self.unique),len(self.unique)) + '100%' + '\n', end='\r')
+        
+    #Export geodataframes, 3 options
+    def create_gdf_premerge(self):
+        # create empty list to fill out
+        myList = []
+        
+        for parts in range(0,len(self.unique)):
+            # get all segment points and make line
+            lineList = _tupleToList(self.unique[parts][0])
+            geom_line = LineString([Point(lineList[0]), Point(lineList[1])])
+            
+            # get other values for premerged
+            UniqueID = parts 
+            Orientation = self.unique[parts][1]
+            linksP1 = self.unique[parts][2]
+            linksP2 = self.unique[parts][3]
+            bestP1 = self.unique[parts][4]
+            bestP2 = self.unique[parts][5]
+            P1Final = self.unique[parts][6]
+            P2Final = self.unique[parts][7]
+            
+            # append list
+            myList.append([UniqueID, Orientation, linksP1, linksP2, bestP1, bestP2, P1Final, P2Final, geom_line])
+
+        edge_gdf = gpd.GeoDataFrame(myList, columns=['UniqueID','Orientation','linksP1','linksP2','bestP1','bestP2','P1Final','P2Final', 'geometry'], crs=self.gdfProjection)
+        edge_gdf.set_index('UniqueID', inplace=True)
+
+        return edge_gdf
+
+    def create_gdf_strokes(self):
+
+        # create empty list to fill 
+        myList = []
+        
+        # loop through merged geometry
+        for a in self.merged:
+
+            # get all segment points and make line strings
+            linelist = _tupleToList(list(self.merged[a]))
+            list_lines_segments = []
+
+            for b in linelist:
+                list_lines_segments.append(LineString(b))      
+            
+            # merge seperate segments
+            geom_multi_line = ops.linemerge(MultiLineString(list_lines_segments))
+
+            # get other values for gdf
+            ID_value = a 
+            nSegments = len(self.merged[a])
+
+            # append list
+            myList.append([ID_value, nSegments, geom_multi_line])
+
+        edge_gdf = gpd.GeoDataFrame(myList, columns=['stroke_group', 'nSegments', 'geometry'], crs=self.gdfProjection)
+        edge_gdf.set_index('stroke_group', inplace=True)
+        
+        return edge_gdf
+
+    def add_gdf_stroke_attributes(self):
+
+        # Invert self.edge_idx to get a dictionary where the key is the original edge index and the value is the group
+        inv_edges = {value: key for key in self.edge_idx for value in self.edge_idx[key]}
+
+        # create empty list that will contain attributes of stroke
+        stroke_group = list()
+
+        for edge in self.uv_index:
+            stroke_group.append(inv_edges[edge])
+        
+        return pd.Series(stroke_group, index=self.edge_gdf.index)
+
+#Set recurrsion depth limit to avoid error at a later stage
+sys.setrecursionlimit(10000)
```

Comment:
I know this comes from the original code but do you know where does it cause an issue? I'd rather not have this here if we can avoid it.

Reply:
@PratyushTripathy do you remember the reason for this?

Reply:
@martinfleis there is a recursion function that reaches maximum depth when the street network is large. While processing it for bigger cities like Delhi, the original code hits the recursion limit, so I had to increase it to be able to process it.

Reply:
@PratyushTripathy Can point to the specific bit in the code where this happens? Just to make a not there.

Reply:
@martinfleis Not able to find it. It's been a while, I don't exactly remember the function. It could be that I re-wrote the function and solved the issue but forgot to remove this line. But to be sure, I'm running a large city again using the old code without increasing the recursion limit. Will get back on this.

Reply:
@martinfleis I processed a large city twice, the first time with that line (recursion limit increase) commented and the second time by setting the limit to 1000 (I guess that's Python's default). It processed easily. Looks like this was something for an older version of code that I missed removing. Please go ahead and remove it. And thanks for flagging :)

Reply:
@PratyushTripathy Brilliant! Thanks for testing it out!